oci/archive/oci_transport.go

```diff
@@ -35,6 +36,7 @@ type ociArchiveReference struct {
 	file         string
 	resolvedFile string
 	image        string
+	sourceIndex  int
```

Comment:
This _must_ show up in `StringWithinTransport`, and the tests should have appropriate coverage.